### DISCUSSION TITLE
리팩토링 기준

### DISCUSSION THREAD

**Discussion by codeisneverodd** [prgrms-fe-devcourse/FEDC2_TimeDivider_Root#29] [0. 개발]
각자 생각하는 리팩토링의 기준을 적어주세요!

**Comment by dahye1013** [comment]
Styled 컴포넌트 어디에 위치 시킬 것인지 
상단 / 하단

**Reply by codeisneverodd** [reply]
덜 중요하니까 아래로 갈까요?

**Reply by dahye1013** [reply]
넵 하단 위치로!

**Comment by dahye1013** [comment]
숫자 및 색상 상수화

**Comment by dahye1013** [comment]
import 이후 개행

**Reply by codeisneverodd** [reply]
쓰지 않는 import 문 삭제

**Comment by codeisneverodd** [comment]
1. 무언가 계산하는 로직이 있다면, 별도의 함수로 분리
2. onClick, onSubmit 등 DOM에 박히는 로직, 별도 함수로 DOM 밖으로 분리

**Comment by codeisneverodd** [comment]
function 되어있는 것 const 형태로 바꾸기?

**Reply by dahye1013** [reply]
일관성 있게 const 선언형 방식으로 했으면 하네요!

**Comment by dahye1013** [comment]
페이지에 대한 라우트 path 정하기

**Reply by codeisneverodd** [reply]
이건 우선 페이지 컴포넌트 이름으로 하는게 좋을 것 같아요!

**Comment by ljw0096** [comment]
console.log pr시 삭제

**Reply by dahye1013** [reply]
앗 이건... 필수!!!

**Comment by codeisneverodd** [comment]
불필요한 주석 삭제

**Comment by dahye1013** [comment]
한 함수에 15줄 미만으로 작성하기

**Reply by codeisneverodd** [reply]
이건 꼭 제한을 하면 좋을까요?

**Reply by dahye1013** [reply]
함수가 너무 많은 역할을 수행하면 다른 사람이 코드를 파악하기 어려울 것 같아요
선언식으로 분리해서 역할을 나누었으면 해요!

**Reply by dahye1013** [reply]
어렵다고 느껴진다면 제한보다는 옵션으로 진행하는게 좋을 것 같네요 ☺️

**Comment by ljw0096** [comment]
Component 디렉토리 정리

많아지니 복잡합니다

**Reply by codeisneverodd** [reply]
요건 개인 리팩토링하고, 다같이 정리하면 좋을 것 같네요!

**Reply by dahye1013** [reply]
역할이 최소화 + 많은 페이지에서 사용하고 있는 경우 `shared` 디렉토리를 만들어서 분리하면 좋겠어요
디렉토리명은 다른거여도 상관없어요 👀 개인 리팩토링 이후 인것도 찬성합니다

**Comment by dahye1013** [comment]
indent(인덴트, 들여쓰기) depth를 3이 넘지 않도록 구현하기

**Reply by codeisneverodd** [reply]
if 같은 중첩문을 말씀하시는 것이군요

**Reply by dahye1013** [reply]
넵 중첩문 반복문 등이 뎁스가 너무 깊어지면 흐름을 파악하기 어려워서요
for문 안에 if문 안에 for문... 이런 경우는 뎁스가 3겹이니까요!
